azure-pipelines.yml

```diff
@@ -25,7 +25,16 @@ steps:
     Write-Host "Target branch: '$(APPVEYOR_REPO_BRANCH)'"
 - task: NuGetToolInstaller@1
   inputs:
-    versionSpec: 
+    versionSpec:
+- task: AzureCLI@2
+  displayName: Login via Azure CLI to acquire access token
+  inputs:
+    azureSubscription: $(AzureSubscriptionName)
+    scriptType: ps
+    scriptLocation: inlineScript
+    inlineScript: |
+      $accessToken = (az account get-access-token --query "accessToken" | % { $_.Trim('"') })
```

Comment:
Do you not need to login first?

Reply:
Good catch, but I think since the AzureCLI task depends on the service connection in Azure DevOps, and the example does not have `az login` command, I think it will login automatically.

Let me check with my private DevOps pipeline tomorrow.

Reply:
@ahmelsayed since the **subscriptionName** parameter accepts a specific service principal. I can confrim that the `az login` is automatically generated. https://dev.azure.com/hazeng0405/guessme/_build/results?buildId=7

Reply:
Got it. Shouldn't you be able to reuse the code here then https://github.com/Azure/azure-functions-core-tools/blob/dev/src/Azure.Functions.Cli/Actions/AzureActions/BaseAzureAction.cs
to get a token?

Reply:
It is a bit unfortunate that the `az` executable will no longer be available outside AzCLI task. So we may still need to export the AccessToken https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=2785&view=logs&jobId=12f1170f-54f2-53f3-20dd-22fc7dff55f9&taskId=036ed663-514a-59f3-0c88-23d2cf182147&lineStart=120&lineEnd=120&colStart=81&colEnd=96

Reply:
I see. that makes sense then. Maybe add that as a comment so we remember that. Thanks Roger!